// formilk/dest/formilk.js
import { Card } from './card.js';
export { Card } from './card.js';
import { InputBasic } from './input-basic.js';
export { InputBasic } from './input-basic.js';
import { InputToggle } from './input-toggle.js';
export { InputToggle } from './input-toggle.js';
import { InputCheckbox } from './input-checkbox.js';
export { InputCheckbox } from './input-checkbox.js';
import { InputRadio } from './input-radio.js';
export { InputRadio } from './input-radio.js';
import { InputColor } from './input-color.js';
export { InputColor } from './input-color.js';
import { Button } from './button.js';
export { Button } from './button.js';
import { Label } from './label.js';
export { Label } from './label.js';
import { Icon } from './icon.js';
export { Icon } from './icon.js';
import 'atomico/jsx-runtime';
import 'atomico';
import '@atomico/hooks/use-slot';
import '@atomico/hooks/use-render';
import './chunks/5ace6c85.js';
import './chunks/7b1738d8.js';
import '@atomico/hooks/use-parent';

customElements.define("formilk-card", Card);
customElements.define("formilk-input-basic", InputBasic);
customElements.define("formilk-input-toggle", InputToggle);
customElements.define("formilk-input-checkbox", InputCheckbox);
customElements.define("formilk-input-radio", InputRadio);
customElements.define("formilk-input-color", InputColor);
customElements.define("formilk-button", Button);
customElements.define("formilk-label", Label);
customElements.define("formilk-icon", Icon);
