// formilk/dest/input-toggle.js
import { jsx } from 'atomico/jsx-runtime';
import { c } from 'atomico';
import { u as useSwitch } from './chunks/7b1738d8.js';
import '@atomico/hooks/use-render';
import '@atomico/hooks/use-parent';
import './chunks/5ace6c85.js';

const style = /*css */ `
    :host{
      display: inline-block;
      margin: .5rem 0px;
    }
    .toggle{
        width: 2rem;
        padding: .25rem;
        border-radius: 5rem;
        cursor: pointer;
        border: .1rem solid var(--input-toggle_border);
        background: var(--input_background);
        box-shadow: var(--input_shadow);
        margin: auto 0px;
        position: relative;
        box-sizing: content-box;
    }
    .fill{
        width: 1rem;
        height: 1rem;
        background: var(--input-toggle_fill);
        border-radius: 100%;
        transition: .2s ease all;
        transform: scale(.75);
    }
    :host([checked]) .fill{
        transform: translateX(100%);
        background: var(--input-toggle_fill-active);
    }
`;

function inputToggle({ name, value }) {
  const [, refContainer, toggle] = useSwitch("checkbox", name, value);

  return (
    jsx('host', { shadowDom: true, toggle: toggle,}
      , jsx('style', null, style)
      , jsx('button', { class: "toggle", ref: refContainer,}
        , jsx('div', { class: "fill",}
          , jsx('slot', null)
        )
      )
    )
  );
}

inputToggle.props = {
  checked: { type: Boolean, reflect: true },
  name: String,
  value: {
    type: null,
    value: "on",
  },
  disabled: { type: Boolean, reflect: true },
};

const InputToggle = c(inputToggle);

export { InputToggle };
